shopinvader_wishlist/models/product_set.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductSet(models.Model):
+
+    _inherit = "product.set"
+
+    shopinvader_backend_id = fields.Many2one(
+        comodel_name="shopinvader.backend",
+        help="If you are using this set for shopinvader customer "
+        "you must select a backend.",
+    )
+    lang_id = fields.Many2one(
+        "res.lang",
+        string="Lang",
+        default=lambda self: self.env["res.lang"]._lang_get(
+            self.env.context.get("lang")
+        ),
+    )
+
+    def get_line_by_product(self, product_id=None, invader_variant_id=None):
+        assert product_id or invader_variant_id
```

Comment:
assert are removed if your code is "compiled" to be optimized. I would prefer a true exception....

Reply:
done and squashed